FileShareService.DesktopClient.Core/JobsParser.cs

```diff
@@ -26,12 +35,144 @@ public Jobs.Jobs Parse(string jobs)
+
+                if (batchJobs == null || batchJobs.Type != JTokenType.Array || !batchJobs.HasValues)
+                {
+                    throw new JsonReaderException("Configuration file formatted incorrectly. Unable to find a job to process.");
+                }
+
+                List<IJob> jobCollection = new List<IJob>();
+
+                jobIdCollection = new List<string>();
+
+                ErrorDeserializingJobsJob? errorJob = null;
+
+                // Deserialize each job and validate
+                foreach (JToken batchJob in batchJobs)
+                {
+                    string jobAction = string.Empty;
+
+                    List<string> errors = ValidateJobActionAndDisplayName(batchJob, out jobAction);
+
+                    if(errors.Any())
+                    {
+                        if(errorJob == null)
+                        {
+                            errorJob = new ErrorDeserializingJobsJob();
+                        }
+                        errorJob.ErrorMessages.AddRange(errors);
+
+                        continue;
+                    }
+
+                    IJob? job = null;
+                    string jsonString = Convert.ToString(batchJob);
+
+                    switch (jobAction)
+                    {
+                        case NewBatchJob.JobAction:
+                            job = JsonConvert.DeserializeObject<NewBatchJob>(jsonString, jsonSerializerSettings);
+                            break;
+
+                        case AppendAclJob.JobAction:
+                            job = JsonConvert.DeserializeObject<AppendAclJob>(jsonString, jsonSerializerSettings);
+                            break;
+
+                        case SetExpiryDateJob.JobAction:
+                            job = JsonConvert.DeserializeObject<SetExpiryDateJob>(jsonString, jsonSerializerSettings);
+                            break;
+                    }
+
+                    if (job != null)
+                    {
+                        job.Validate(batchJob);
+                        jobCollection.Add(job);
+                    }
+                }
+
+                //Add error job at begining of the collection, if exists.
+                if (errorJob != null)
+                {
+                    jobCollection.Insert(0, errorJob);
+                }
+
+                return  new Jobs.Jobs() { jobs = jobCollection};
             }
             catch (Exception e)
             {
-                return new Jobs.Jobs() {jobs = new[] {new ErrorDeserializingJobsJob(e)}};
+                return new Jobs.Jobs() { jobs = new[] { new ErrorDeserializingJobsJob(e) { ErrorMessages = new List<string>() { e.Message } } } };
+            }
+        }
+
+        private List<string> ValidateJobActionAndDisplayName(JToken job, out string jobAction)
+        {
+            jobAction = string.Empty;
+
+            List<string> errors = new List<string>();
```

Comment:
Modified the implementation to return more than one error. If job action/display name is null or empty, then it will return single error message. Otherwise, it will return error collection. So, it will return List<string>